modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -814,6 +821,16 @@ private void updateAssignmentInternal(ConfigurationNotificationEvent<byte[]> ass
         CompletableFuture.allOf(futures).join();
     }
 
+    private SnapshotAwareMvPartitionStorage getOrCreateSnapshotAwareMvPartition(InternalTable internalTbl, int partId) {
+        MvPartitionStorage storage = getOrCreateMvPartition(internalTbl, partId);
+
+        return new SnapshotAwareMvPartitionStorage(storage, outgoingSnapshotsManager, new PartitionKey(internalTbl.tableId(), partId));
```

Comment:
This method shuld return an existing instance, if it is already created, based on the name (getOrCreate). But it always returns a new one.

Reply:
Fixed this